### PR TITLE
Feature merchant all customers

### PR DESCRIPTION
**Pull request by CCirbo**
Created a merchant_customers_spec.rb test file with tests for the GET /api/v1/merchants/:merchant_id/customers endpoint.
Added a MerchantCustomersController to handle retrieving all customers for a specific merchant.
Added a joins method to the Customer model to handle the logic for fetching customers associated with a merchant.
Updated the routes to include a new route for GET /api/v1/merchants/:merchant_id/customers to return all customers for a given merchant.
Created a customer serializer file

### CODE REVIEW

File: app/models/customer.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
 class Customer < ApplicationRecord
     has_many :invoices
+    has_many :merchants, through: :invoices
+
+    def self.for_merchant(merchant_id)
+        joins(:invoices).where(invoices: { merchant_id: merchant_id }).distinct
```
Comment (TDManning) [inline app/models/customer.rb:6]: Nice work incorporating joins here!

**Review by TDManning — COMMENTED** [review]
Nice refactors based on the before/each block.

**Review by TDManning — COMMENTED** [review]
This all looks good!

**Comment by jimmacur** [comment]
Great job on this update! I appreciate the use of the joins method in the for_merchant scope, which simplifies the query and ensures we’re only fetching distinct customers for a given merchant. The test coverage looks thorough, especially for edge cases like merchants with no customers. Everything seems well structured! Nice job!

**Comment by MDelarosa1993** [comment]
Thanks for the detailed work on this. It enhances our API functionality and maintains good test coverage. I’ll review it more closely, but everything seems well-implemented so far!
